src/driver/sqlserver/SqlServerDriver.ts

```diff
@@ -434,7 +434,8 @@ export class SqlServerDriver implements Driver {
             return "uniqueidentifier";
 
         } else if (column.type === "simple-array" || column.type === "simple-json") {
-            return "ntext";
+            column.length = "MAX";
```

Comment:
I dont think so this is right place to modify `column.length`. I am not 100% sure but this should be moved to `getColumnLength` method of driver.

Reply:
this is definitely bad place for column length assignment to be. Also its not clear what `column` is in this method. Those places were developed by @AlexMesser and unfortunately I don't know most of the details.

Reply:
@leoiii12 we already have similar logic for uuid in mysql. We have code [here](https://github.com/pleerock/typeorm/blob/7922f12da3cba3d6605929a7fea93cdf88c2d98e/src/driver/mysql/MysqlDriver.ts#L716) and [here](https://github.com/typeorm/typeorm/blob/7922f12da3cba3d6605929a7fea93cdf88c2d98e/src/driver/mysql/MysqlDriver.ts#L585). Can't we do same in this case?

Reply:
Problem is with `getColumnLength` where first argument is possible `TableColumn` and this object contains only normalized data type (database data type).